src/main/java/net/datafaker/providers/base/Compass.java

```diff
@@ -0,0 +1,110 @@
+package net.datafaker.providers.base;
+
+/**
+ * @since 1.7.0
+ */
+public class Compass extends AbstractProvider<BaseProviders> {
+
+    enum Point { CARDINAL, ORDINAL, HALF_WIND, QUARTER_WIND }
+
+    private Point point;
```

Comment:
Then how about `Direction`?

Reply:
Would be great if you first a bit elaborate the meaning of this enum and what for it's intended to use

Reply:
The compass rose has different directions, such as Cardinal, Ordinal, Half wind and Quarter wind. Cardinal is four main points of a compass: north, south, east, and west. Ordinal directions refer to the direction found at the point equally between each cardinal direction, and so on. 

I introduced an enum in order to not have a lot of methods like this:
```
   public String cardinalDirectionWord() {
        return resolve("compass.cardinal.word");
    }

    public String cardinalDirectionAbbreviation() {
        return resolve("compass.cardinal.abbreviation");
    }

    public String cardinalDirectionAzimuth() {
        return resolve("compass.cardinal.azimuth");
    }

    public String ordinalDirectionWord() {
        return resolve("compass.ordinal.word");
    }

    public String ordinalDirectionAbbreviation() {
        return resolve("compass.ordinal.abbreviation");
    }

    public String ordinalDirectionAzimuth() {
        return resolve("compass.ordinal.azimuth");
    }
```

And instead of this, just specify the direction. I hope I've explained my point of view.